libraries/botbuilder/src/skills/skillHandlerImpl.ts

```diff
@@ -0,0 +1,212 @@
+            if (this.adapter instanceof BotFrameworkAdapter) {
+                const client = this.adapter.createConnectorClient(newActivity.serviceUrl);
+                context.turnState.set(this.adapter.ConnectorClientKey, client);
+            }
+
+            switch (newActivity.type) {
+                case ActivityTypes.EndOfConversation:
+                    await this.conversationIdFactory.deleteConversationReference(conversationId);
+
+                    this.applySkillActivityToTurnContext(context, newActivity);
+                    await this.logic(context);
+
+                    break;
+
+                case ActivityTypes.Event:
+                    this.applySkillActivityToTurnContext(context, newActivity);
+                    await this.logic(context);
+
+                    break;
+
+                default:
+                    resourceResponse = await context.sendActivity(newActivity);
+                    break;
+            }
+        });
+
+        if (!resourceResponse) {
+            resourceResponse = { id: uuid() };
+        }
+
+        return resourceResponse;
+    }
+
+    private async continueConversation(
+        claimsIdentity: ClaimsIdentity,
+        conversationId: string,
+        callback: (context: TurnContext, ref: SkillConversationReference) => Promise<void>
+    ): Promise<void> {
+        const ref = await this.getSkillConversationReference(conversationId);
+
+        const continueCallback = (context: TurnContext): Promise<void> => {
+            context.turnState.set(this.skillConversationReferenceKey, ref);
+            return callback(context, ref);
+        };
+
+        try {
+            await this.adapter.continueConversationWithClaims(
+                claimsIdentity,
+                ref.conversationReference,
+                ref.oAuthScope,
+                continueCallback
+            );
+        } catch (err) {
+            if (err.message === 'NotImplemented') {
+                // We're in the legacy scenario where our adapter does not support passing through claims/audience
+                // explicitly. Stash it in turn context and hope for the best!
+                await this.adapter.continueConversation(ref.conversationReference, async (context) => {
+                    context.turnState.set(context.adapter.BotIdentityKey, claimsIdentity);
+                    return continueCallback(context);
+                });
+            } else {
+                throw err;
+            }
+        }
+    }
+
+    // adapter.continueConversation() sends an event activity with continueConversation in the name.
+    // this warms up the incoming middlewares but once that's done and we hit the custom callback,
+    // we need to swap the values back to the ones received from the skill so the bot gets the actual activity.
+    private applySkillActivityToTurnContext(context: TurnContext, activity: Partial<Activity>): void {
```

Comment:
https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Skills/SkillHandlerImpl.cs#L94

Only thing missing is `Properties`, which I think it specific to .NET.